### ISSUE TITLE
feat(validator): add validation support for `RadioGroup`

### ISSUE THREAD

**Issue by Icyscools** [ayoayco/astro-reactive-library#230]
Right now, `RadioGroup` don't have validation support from our validator package. You can try with this code.
```js
const form = new FormGroup([{
  name: "rating",
  label: "Rating",
  type: "radio",
  validators: [Validators.required],
  triggerValidationOn: "click",
  options: ["1", "2", "3", "4", "5"],
}]);
---
<Form showValidationHints={true} validateOnLoad={true} formGroups={form} />
```
Currently, the result will look like this

![image](https://user-images.githubusercontent.com/30778789/205911694-4816c938-1dba-4463-b67e-2ea1cab62620.png)

The `RadioGroup` can't validate required the options after we select some options. The goal for this issue is to have a `RadioGroup` with no validation error after selecting some option in `RadioGroup` and re-validate from client-side.

For further investigation, we have some discussion on #224. Please, have a look in there.

**Comment by ayoayco** [comment]
Ah I see, yeah validating the RadioGroup will need a special handling I believe since we treat it as one FormControl, when under the hood it is really multiple input elements.

Also, other validators may be used, like `Validators.min(8)` which means they want the value to be great than 8. In this case, we need to make sure RadioGroup.value is set correctly and is being checked instead of each of the RadioOption.

**Comment by alexsam29** [comment]
Since you said we treat RadioGroup as one FormControl I decided to compare it with the other Controls and see why it's acting different. I noticed that the other controls only contain `<input>` and `<label>` elements. So, curiously I removed the `<div class="radio-option">` element from the RadioGroup in `RadioGroup.astro`  and just did this:
```js
options.map((option: ControlOption, index: number) => (
<>
	<input
		type="radio"
		id={control.id + '-' + index}
		name={control.name}
		value={option.value}
		checked={option.value === control.value}
		readonly={readOnly || null}
		disabled={readOnly || null}
		data-validation-on={control.triggerValidationOn ? control.triggerValidationOn : null}
	/>
	<label for={control.id + '-' + index}>{option.label}</label>
	<br />
</>
))
```

Now when I select a radio button created with the source code above, the error goes away when you select any one of the radio buttons.

<img width="159" alt="image" src="https://user-images.githubusercontent.com/69481177/206827903-a245cd0f-d6ee-4579-b5c4-0deaa57d5ae3.png">

I'm not too sure if this completely gets rid of this issue, but thought it was worth mentioning.
